custom_components/opnsense/migrate.py

```
@@ -0,0 +1,470 @@
+    new_entry_bool = hass.config_entries.async_update_entry(
+        config_entry, data=new_data, unique_id=new_device_unique_id, version=3
+    )
+    if new_entry_bool:
+        _LOGGER.debug("[migrate_2_to_3] config_entry update successful")
+    else:
+        _LOGGER.error("Migration of config_entry to version 3 unsuccessful")
+        return False
+    return True
+
+
+async def _migrate_3_to_4(
```

Ruff (0.15.21)

[warning] Too many branches (20 > 12)

(PLR0912)